### PR TITLE
CM-468: Iterating over JSON Object in C#

### PR DESCRIPTION
**Pull request by aineaTheSecond**

### CODE REVIEW

File: json-csharp/IteratingOverJsonObject/IteratingOverJsonObject.Tests/IteratingOverJsonObject.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (karimkod) [inline json-csharp/IteratingOverJsonObject/IteratingOverJsonObject.Tests/IteratingOverJsonObject.Tests.csproj:4]: We should be targeting .Net 7

File: json-csharp/IteratingOverJsonObject/IteratingOverJsonObject/IteratingOverJsonObject.csproj [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (karimkod) [inline json-csharp/IteratingOverJsonObject/IteratingOverJsonObject/IteratingOverJsonObject.csproj:5]: Same here, upgrade to .Net 7

File: json-csharp/IteratingOverJsonObject/IteratingOverJsonObject/JsonIteration.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+namespace IteratingOverJsonObject;
+
+public class JsonIteration
+{
+    public string Json { get; set; }
+    public string Name { get; set; }
+    public int Age { get; set; }
+    public string Department { get; set; }
+    public int ItemsCount { get; set; }
```
Comment (karimkod) [inline json-csharp/IteratingOverJsonObject/IteratingOverJsonObject/JsonIteration.cs:14]: As suggested in the article review, these variables don't add much value and they can be local variables to each method (and we can use that to print)

File: json-csharp/IteratingOverJsonObject/IteratingOverJsonObject/JsonIteration.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    {
+        var jsonData = JsonConvert.DeserializeObject<dynamic>(jsonString);
+
+        foreach (var data in jsonData)
+        {
+            Name = data.name;
+            Age = data.age;
+            Department = data.department;
+            ItemsCount += 1;
+        }
```
Comment (karimkod) [inline json-csharp/IteratingOverJsonObject/IteratingOverJsonObject/JsonIteration.cs:31]: The foreach of all methods don't do much, maybe we can do the printing here

File: json-csharp/IteratingOverJsonObject/IteratingOverJsonObject/JsonIteration.cs [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    public int Age { get; set; }
+    public string Department { get; set; }
+    public int ItemsCount { get; set; }
+
+    public JsonIteration()
+    {
+        Json = new TestData().GenerateJsonData();
+    }
+
+    public void IterateOverJsonDynamically(string jsonString)
```
Comment (karimkod) [inline json-csharp/IteratingOverJsonObject/IteratingOverJsonObject/JsonIteration.cs:21]: We can remove the jsonString argument and use the Json property in this class

File: json-csharp/IteratingOverJsonObject/IteratingOverJsonObject/TestData.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+                ""name"":""Ramiro Buck"",
+                ""age"": 28,
+                ""department"":""IT""
+            },
+            {
+                ""name"":""Shirley Pearson"",
+                ""age"": 30,
+                ""department"":""Human Resource""
+            }
+        ]";
```
Comment (karimkod) [inline json-csharp/IteratingOverJsonObject/IteratingOverJsonObject/TestData.cs:33]: We can use a string literal with triple quotes (""") that was introduced in .Net 7 for better readability.

**Review by karimkod — CHANGES_REQUESTED** [review]

**Comment by aineaTheSecond** [comment]
Hello @karimkod, I've added the changes to the source code as requested. Please take a look and share your feedback. 
Changes in the article are underway, I'll ping you once they are ready.

File: json-csharp/IteratingOverJsonObject/IteratingOverJsonObject/JsonIteration.cs [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+            var department = (string)data["department"];
+
+            Console.WriteLine("Name: {0}, Age: {1}, Department: {2} \n", name, age, department);
+        }
+
+        var count = jsonArray.Count;
+        return count;
+    }
+
+    public List<Employee> IterateUsingDynamicallyTypedObject()
```
Comment (karimkod) [inline json-csharp/IteratingOverJsonObject/IteratingOverJsonObject/JsonIteration.cs:51]: This should be named `IterateUsingStaticallyTypedObject`

**Review by karimkod — CHANGES_REQUESTED** [review]
Hey @aineaTheSecond just one small change and I think is ready

File: json-csharp/IteratingOverJsonObject/IteratingOverJsonObject/JsonIteration.cs (hunk 7 above)
Comment (aineaTheSecond) [reply json-csharp/IteratingOverJsonObject/IteratingOverJsonObject/JsonIteration.cs]: This is also done, please take a look

**Comment by karimkod** [comment]
retest this please

**Review by karimkod — APPROVED** [review]

**Pull request merged**